File: hmode/shared/golden-repos/typescript-nodejs-cli/src/commands/greet.ts
/**
 * Greet command - example simple command.
 */

import { Command } from 'commander';
import chalk from 'chalk';

export const greetCommand = new Command('greet')
  .description('Greet a user')
  .argument('<name>', 'Name to greet')
  .option('-l, --loud', 'Greet loudly')
  .action((name: string, options: { loud?: boolean }) => {
    const greeting = `Hello, ${name}!`;

    if (options.loud) {
      console.log(chalk.bold.green(greeting.toUpperCase()));
    } else {
      console.log(chalk.green(greeting));
    }
  });

File: shared/golden-repos/python-cli/src/commands/greet.py
"""Greet command - example simple command."""

import click
from rich.console import Console
from rich.panel import Panel

console = Console()


@click.command()
@click.argument("name")
@click.option("-l", "--loud", is_flag=True, help="Greet loudly")
@click.pass_context
def greet(ctx: click.Context, name: str, loud: bool) -> None:
    """Greet a user by NAME."""
    greeting = f"Hello, {name}!"

    if loud:
        console.print(Panel(greeting.upper(), style="bold green"))
    else:
        console.print(f"[green]{greeting}[/green]")
